x/evm/abci.go

```diff
@@ -444,8 +444,10 @@ func handleConfirmedEvents(ctx sdk.Context, bk types.BaseKeeper, n types.Nexus,
 			continue
 		}
 
+		handledEvents := int64(0)
 		var event types.Event
-		for queue.DequeueUntil(&event, shouldHandleEvent) {
+		for handledEvents < ck.GetParams(ctx).EndBlockerLimit && queue.DequeueUntil(&event, shouldHandleEvent) {
```

Comment:
```suggestion
		endBlockerLimit := ck.GetParams(ctx).EndBlockerLimit
		for handledEvents < endBlockerLimit && queue.DequeueUntil(&event, shouldHandleEvent) {
```
Don't need to hit the keeper every time